PolyEngine/Engine/Src/Engine.cpp

```diff
@@ -2,34 +2,89 @@
 
 using namespace Poly;
 
-Engine::Engine(IGame* game) : Game(game) {
+Engine::Engine(IGame* game) : Game(game) 
+{
 	Game->RegisterEngine(this);
 	Renderer = CreateRenderingContext();
 }
 
-bool Engine::Init(const IRenderingContextParams* context) {
+bool Engine::Init(const IRenderingContextParams* context) 
+{
 	if (!Renderer->Init(context))
 		return false;
 	Game->Init();
 	return true;
 }
 
-void Poly::Engine::Deinit()
+void Engine::Deinit()
 {
 	Renderer->Deinit();
 	delete Renderer;
 	Renderer = nullptr;
 }
 
+void Engine::RegisterPreupdatePhase(PhaseUpdateFunction phaseFunction)
```

Comment:
Let's pass these functions by reference to avoid double copy (1. on call, 2. on PushBack)